libsolidity/interface/CompilerStack.cpp

```diff
@@ -747,22 +747,28 @@ void CompilerStack::compileContract(
 	}
 }
 
+string const CompilerStack::lastContractName() const
+{
+	if (m_contracts.empty())
+		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("No compiled contracts found."));
+	// try to find some user-supplied contract
+	string contractName;
+	for (auto const& it: m_sources)
+		for (ASTPointer<ASTNode> const& node: it.second.ast->nodes())
+			if (auto contract = dynamic_cast<ContractDefinition const*>(node.get()))
+				contractName = contract->fullyQualifiedName();
+	return contractName;
```

Comment:
Maybe you want to assert `contractName` is not empty.

Reply:
Actually I think it is not necessary because in that case `lastContractName` cannot be used to determine if a contract was compiled and `contract()` will fail anyway, because `""` is (hopefully) an impossible contract name :)